packages/datadog-plugin-rhea/src/index.js

```diff
@@ -0,0 +1,215 @@
+'use strict'
+
+const dd = Symbol('datadog')
+const circularBufferConstructor = Symbol('circularBufferConstructor')
+
+function createWrapSend (tracer, config, instrumenter) {
+  return function wrapSend (send) {
+    return function sendWithTrace (msg, tag, format) {
+      patchCircularBuffer(this, instrumenter)
```

Comment:
Instead of doing this lazily, would it be possible to use `defineProperty` on the prototype of `Session` so that it's patched as soon as it's assigned in the constructor?